### PR TITLE
Fixed persistent DB creation is not atomic (fixes SALTO-1377)

### PR DESCRIPTION
**Pull request by roironn**
_Added connection close in vscode and improved db creation_

---

_The code that creates the persistent db connection was not atomic, causing a failure when two remote maps when the same location were created in the same time_

---
_Release Notes_: 
_NA_

### CODE REVIEW

**Comment by yanaise** [comment]
Please change the contents of "Release Notes"

**Review by yanaise — APPROVED** [review]

File: packages/vscode/src/extension.ts [hunk 1]
```diff
@@ -112,3 +112,7 @@ export const activate = async (context: vscode.ExtensionContext): Promise<void>
   },
   async () => onActivate(context))
 )
+
+export const deactviate = async (): Promise<void> => {
```
Comment (netama) [inline packages/vscode/src/extension.ts:116]: nit: `deactviate` -> `deactivate`

File: packages/core/src/local-workspace/remote_map.ts [hunk 2]
```diff
@@ -491,15 +491,16 @@ remoteMap.RemoteMapCreator => {
...
         throw new Error('Failed to open rocksdb connection - too much open connections already')
       }
-      await createDBIfNotExist(location)
-      const readOnly = !persistent
-      const connection = getOpenDBConnection(location, readOnly)
-      persistentDB = await connection
-
-      currnetConnectionsCount += 2
+      const connectionPromise = (async () => {
+        currnetConnectionsCount += 2
```
Comment (netama) [inline packages/core/src/local-workspace/remote_map.ts:495]: nit: `currnet` -> `current`

File: packages/core/src/local-workspace/remote_map.ts [hunk 3]
```diff
@@ -491,15 +491,16 @@ remoteMap.RemoteMapCreator => {
...
+        await createDBIfNotExist(location)
+        const readOnly = !persistent
+        return getOpenDBConnection(location, readOnly)
+      })()
```
Comment (ori-moisis) [inline packages/core/src/local-workspace/remote_map.ts:499]: style nit - this is not a very conventional pattern in our codebase. we usually don't define a function and call it in place.

**Review by ori-moisis — COMMENTED** [review]
1. if this is supposed to fix a bug, please link it to the relevant ticket
2. if this is supposed to fix a bug, there should be a test about it
3. can you please add a more readable title for the PR?

**Pull request merged**